### PR TITLE
Switch to custom python2-qt5-devel for RHEL

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Developers should now use `python2-qt5-devel` to avoid future clashes with EPEL updates.

**To test:**

Code review

*This does not require release notes* because **it is a developer problem.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
I haven't tried to install it yet, but you missed `python36-qt5-devel` in the `%if 0%{?el7}` near the bottom. RHEL7 doesn't define `with_python3` and has unique package names.

**Comment by martyngigg** [comment]
I hadn't changed the el7 as it looked like only framework packages. Shall I add the rest for python36 or should we wait until we have sorted matplotlib out?

**Comment by peterfpeterson** [comment]
I'd prefer having everything we can for RHEL7

File: buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec [hunk 1]
```diff
@@ -111,11 +110,18 @@ Requires: boost-python3-devel
...
 Requires: python36-numpy
+Requires: python36-scipy
+Requires: python36-sphinx
+Requires: python36-dateutil
+Requires: python36-h5py
 Requires: python36-PyYAML
+Requires: python36-mock
+Requires: python36-psutil
+Requires: python36-requests
+Requires: boost-python36-devel
```
Comment (peterfpeterson) [inline buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec]: Can we add `boost169-python2-devel` and `boost169-python3-devel`  to this list?

Comment (martyngigg) [reply buildconfig/dev-packages/rpm/mantid-developer/mantid-developer.spec]: :+1:

**Comment by martyngigg** [comment]
I've built a test version in my [copr sandbox](https://copr-be.cloud.fedoraproject.org/results/martyngigg/sandbox/epel-7-x86_64/01087706-mantid-developer/mantid-developer-1.34-1.el7.noarch.rpm)

**Review by peterfpeterson — APPROVED** [review]
The changes look correct and the package installs on my system. It added `python36-mock`, `python36-pbr`, and `python36-psutil`.

**Comment by martyngigg** [comment]
@mantidproject/gatekeepers The build failure is unrelated here. Can we get this merged for the nightly so the rpm has the correct `python2-qt5` dependency?

**Pull request merged**